### PR TITLE
bug fix: allow all authenticated users to sign out

### PR DESCRIPTION
**Pull request by suyangyu**
# Pull Request Description
## List the changes made in comparison to main:
* bug fix to allow users without an enrolment to sign out

## Jira ticket(s):
* no jira

## Checklist
* [ ] Have you consulted with a QA?
    * [ ] Tick if you expect this work could break the existing Acceptance Test
* [x] Is the branch up to date with main?
* [x] Have you added tests for any changed functionality?
* [x] Have you run the unit test?
* [x] Have you run the it/test?
* [ ] Have you run the [senior-accounting-officer-acceptance-tests](https://github.com/hmrc/senior-accounting-officer-acceptance-tests) suite `run_all_tests.sh`?
* [x] Have you formatted the new/updated Scala sources using `sbt lint`?
* [ ] Does this work need to be coordinated with any other work currently in flight?
  * If ticked please state them below

### CODE REVIEW

File: it/test/controllers/actions/NoEnrolmentRequiredActionISpec.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
```
Comment (dandaviesbrackett) [inline it/test/controllers/actions/NoEnrolmentRequiredActionISpec.scala]:
```suggestion
 * Copyright 2026 HM Revenue & Customs
```

**Review by dandaviesbrackett — DISMISSED** [review]

File: app/controllers/auth/AuthController.scala
Comment (lxol) [inline app/controllers/auth/AuthController.scala]: Why do we need to have any identity verification here at all? 
what's wrong with something like 
```
def singOut ... = Action { implicit request =>
Redirect(config.signOutUrl, Map("continue" -> Seq(config.exitSurveyUrl)))
  }
```
anyone could singout then. Or I missed something?

Comment (suyangyu) [reply app/controllers/auth/AuthController.scala]: we could, though you should be signed in in order to use sign out, so then there exist an extreme edge case when a user not signed in hitting it and it breaks.

File: test/controllers/actions/FakeNoEnrolmentRequiredAction.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
```
Comment (jacob-leeming-netcompany) [inline test/controllers/actions/FakeNoEnrolmentRequiredAction.scala]: another 2025

File: app/controllers/auth/AuthController.scala
Comment (lxol) [reply app/controllers/auth/AuthController.scala]: But Redirect doesn't use anything from the request. Would it actually break for not signed-in users?

**Review by lxol — APPROVED** [review]

**Review by dandaviesbrackett — APPROVED** [review]
